### PR TITLE
Feature/pretty print

### PR DESCRIPTION
**Pull request by fordie**
*requires acceptance tests from feature/pretty-print*

All tests pass, Tarakul is happy with my demo

### CODE REVIEW

File: app/uk/gov/hmrc/offpayroll/controllers/PrintController.scala [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        "completedBy" -> text,
+        "client" -> text,
+        "job" -> text,
+        "reference" -> text
+      )(PrintResult.apply)(PrintResult.unapply)
+    )
+
+    printPrint.bindFromRequest.fold (
+      formWithErrors => {
+        throw new Exception("Hidden fields missing from the form") // fixme make this the correct type of Exception
```
Comment (peteanning) [inline app/uk/gov/hmrc/offpayroll/controllers/PrintController.scala]: This needs to match L48

File: app/uk/gov/hmrc/offpayroll/views/interview/display_decision.scala.html [hunk 2]
```diff
@@ -9,11 +9,12 @@
 @import uk.gov.hmrc.offpayroll.models.Decision
 @import uk.gov.hmrc.offpayroll.models.{IN, OUT, UNKNOWN}
 
-@(decision: Decision, interview: List[(String, List[String])], esi: Boolean)(implicit request: Request[_], messages: Messages)
+@import uk.gov.hmrc.offpayroll.util.CompressedInterview
```
Comment (peteanning) [inline app/uk/gov/hmrc/offpayroll/views/interview/display_decision.scala.html]: I don't think we need the compressed interview import on L12

File: test/uk/gov/hmrc/offpayroll/controllers/PrintControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      val maybeRoute = route(fakeApplication, FakeRequest(POST, "/check-employment-status-for-tax/print/print").withFormUrlEncodedBody(HIDDEN_FIELDS_AND_FORM.toList :_*))
+      maybeRoute.isDefined shouldBe true
+      maybeRoute.map { route =>
+        val result = await(route)
+        status(result) shouldBe Status.OK
+      }
+    }
+  }
+
+//  "GET /start" should {
```
Comment (peteanning) [inline test/uk/gov/hmrc/offpayroll/controllers/PrintControllerSpec.scala]: Please remove all commented out code

File: test/uk/gov/hmrc/offpayroll/controllers/PrintControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    }
+  }
+  "POST /print/format" should {
+    "return 500 and report missing fields" in {
+      val fakeRequest = FakeRequest(POST, "/check-employment-status-for-tax/print/format").withFormUrlEncodedBody("esi" -> "false")
+      intercept[IllegalStateException]{new PrintController().format()(fakeRequest).futureValue}
+    }
+  }
+  "POST /print/print" should {
+    "return 200 and didplay pretty print result" in {
```
Comment (peteanning) [inline test/uk/gov/hmrc/offpayroll/controllers/PrintControllerSpec.scala]: Should be testing the unhappy path as above

**Comment by fordie** [comment]
fixed & re-ran tests, which all pass with flying colours

**Pull request merged**